### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by ch3njust1n**
# Description

Unit tests for:
1. `/login/username`
2. `/login/email`
3. `/signup`
4. `/recover/forgot`

## Checklist

- [x] This PR can be reviewed in under 30 minutes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have assigned reviewers to this PR.

### CODE REVIEW

File: packages/server/src/auth/auth.service.ts [hunk 1]
```diff
@@ -59,9 +59,13 @@ export class AuthService {
    * @param projectId
    * @param email
    */
-  forgotPassword(projectId: string, email: string): void {
+  async forgotPassword(projectId: string, email: string): Promise<void> {
     const resetCodePlain = randomstring.generate(10);
-    this.userService.setResetToken(projectId, email, resetCodePlain);
+    const wasSet = await this.userService.setResetToken(projectId, email, resetCodePlain);
+    console.log(wasSet);
```
Comment (cbolles) [inline packages/server/src/auth/auth.service.ts]: NIT: Remove extra `console.log`

File: packages/server/src/auth/auth.service.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class AuthService {
    * @returns JWT or 401 status code
    */
   async validateUsername(projectId: string, username: string, password: string): Promise<any> {
-    if (projectId != null) {
+    if (projectId != null && username != null && password != null) {
```
Comment (cbolles) [inline packages/server/src/auth/auth.service.ts]: Something that may be an improvement. In the case where the arguments aren't provided, maybe have the status code be an `HttpStatus.BAD_REQUEST`. That ways users of the auth service can more easily debug their code.

**Review by cbolles — CHANGES_REQUESTED** [review]
Looks good, one possible suggestion and one spare print to remove

**Review by cbolles — CHANGES_REQUESTED** [review]
Looks good, one possible suggestion and one spare print to remove

**Review by cbolles — DISMISSED** [review]

File: packages/server/src/auth/auth.service.ts (hunk 2 above)
Comment (cbolles) [reply packages/server/src/auth/auth.service.ts]: So my thought here was that `BAD_REQUEST` is returned if something is wrong with the format of the data and `UNAUTHORIZED` when authorization specifically fails. That way is easier to find out if the issue is genuinely the auth service not allowing access or if the issue is a malformed request

**Review by cbolles — COMMENTED** [review]
Looks good, have a clarification on my thought on the HTTP exception

File: packages/server/src/auth/auth.service.ts (hunk 2 above)
Comment (ch3njust1n) [reply packages/server/src/auth/auth.service.ts]: Ah ok thanks for the clarification. That makes sense.

**Review by cbolles — APPROVED** [review]

**Pull request merged**